src/libraries/Microsoft.PowerFx.Core/Functions/UserDefinedFunction.cs

```diff
@@ -182,15 +183,20 @@ public UserDefinedFunction WithBinding(INameResolver nameResolver, IBinderGlue b
             binding = func.BindBody(nameResolver, binderGlue, bindingConfig, features, rule);
 
             return func;
-        }
-
-        private static readonly ISet<string> _restrictedUDFNames = new HashSet<string> { "Type", "IsType", "AsType" };
+        }
+
+        // Adding a restricted UDF name is a breaking change, this test will need to be updated and a conversion will be needed for existing scenarios
+        private static readonly ISet<string> _restrictedUDFNames = new HashSet<string>
+        {
+            "Type", "IsType", "AsType", "Set", "Collect", "ClearCollect",
+            "UpdateContext", "Navigate", "Select", "Reset",

```

Comment:
>"Select", "Reset" [](http://example.com/codeflow?start=41&length=17)

Curious: why are Select and Reset restricted? Does analysis use these functions for something special?